### ISSUE TITLE
pytest-cov 2.5 doesn't seem to work with xdist

### ISSUE THREAD

**Issue by tgs** [pytest-dev/pytest-cov#157]
It seems to cause an endless loop of restarting the workers, which immediately fail each time.

Steps to reproduce:
```
virtualenv -p python3.4 env
source env/bin/activate
pip install pytest pytest-cov==2.5 pytest-xdist
cat > test_things.py <<END
def test_a():
    assert '1' == '1'

def test_b():
    assert '2' == '2'
END
py.test test_things.py  --dist=load --tx="2*popen//python=`which python`" -v --cov --cov-report=term
```

This succeeds with pytest-cov 2.4, but fails with 2.5.  The backtrace is something like this:

```
=================================== test session starts ===================================
platform linux -- Python 3.4.5, pytest-3.0.7, py-1.4.33, pluggy-0.4.0 -- /home/smithtg/.virtualenvs/ptc/bin/python3.4
cachedir: .cache
rootdir: /home/smithtg/.virtualenvs/ptc, inifile:
plugins: xdist-1.16.0, cov-2.5.0
[gw0] linux Python 3.4.5 cwd: /home/smithtg/.virtualenvs/ptc
[gw1] linux Python 3.4.5 cwd: /home/smithtg/.virtualenvs/ptc
gw0 C / gw1 CTraceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1539, in <module>
  File "<string>", line 1534, in serve
  File "<string>", line 1047, in serve
  File "<string>", line 259, in integrate_as_primary_thread
  File "<string>", line 277, in _perform_spawn
  File "<string>", line 219, in run
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 585, in _should_trace
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 530, in _should_trace_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 551, in _check_include_omit_etc_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/files.py", line 242, in match
TypeError: startswith first arg must be str or a tuple of str, not bool
Exception ignored in: Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 585, in _should_trace
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 530, in _should_trace_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 551, in _check_include_omit_etc_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/files.py", line 242, in match
TypeError: startswith first arg must be str or a tuple of str, not bool
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 585, in _should_trace
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 530, in _should_trace_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 551, in _check_include_omit_etc_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/files.py", line 242, in match
TypeError: startswith first arg must be str or a tuple of str, not bool
[gw0] node down: Not properly terminated
Replacing crashed slave gw0
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1539, in <module>
  File "<string>", line 1534, in serve
  File "<string>", line 1047, in serve
  File "<string>", line 259, in integrate_as_primary_thread
[gw2] linux Python 3.4.5 cwd: /home/smithtg/.virtualenvs/ptc
gw2 C / gw1 C  File "<string>", line 277, in _perform_spawn
  File "<string>", line 219, in run
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 585, in _should_trace
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 530, in _should_trace_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 551, in _check_include_omit_etc_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/files.py", line 242, in match
TypeError: startswith first arg must be str or a tuple of str, not bool
Exception ignored in: Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 585, in _should_trace
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 530, in _should_trace_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 551, in _check_include_omit_etc_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/files.py", line 242, in match
TypeError: startswith first arg must be str or a tuple of str, not bool
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 585, in _should_trace
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 530, in _should_trace_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 551, in _check_include_omit_etc_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/files.py", line 242, in match
TypeError: startswith first arg must be str or a tuple of str, not bool
[gw1] node down: Not properly terminated
Replacing crashed slave gw1
[gw3] linux Python 3.4.5 cwd: /home/smithtg/.virtualenvs/ptc
gw2 C / gw3 CTraceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1539, in <module>
  File "<string>", line 1534, in serve
  File "<string>", line 1047, in serve
  File "<string>", line 259, in integrate_as_primary_thread
  File "<string>", line 277, in _perform_spawn
  File "<string>", line 219, in run
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 585, in _should_trace
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 530, in _should_trace_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 551, in _check_include_omit_etc_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/files.py", line 242, in match
TypeError: startswith first arg must be str or a tuple of str, not bool
Exception ignored in: Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 585, in _should_trace
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 530, in _should_trace_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 551, in _check_include_omit_etc_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/files.py", line 242, in match
TypeError: startswith first arg must be str or a tuple of str, not bool
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 585, in _should_trace
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 530, in _should_trace_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/control.py", line 551, in _check_include_omit_etc_internal
  File "/home/smithtg/.virtualenvs/ptc/lib/python3.4/site-packages/coverage/files.py", line 242, in match
TypeError: startswith first arg must be str or a tuple of str, not bool
[gw2] node down: Not properly terminated
```

**Comment by tgs** [comment]
Another detail -

`py.test test_things.py  --dist=load --tx="2*popen//python=`which python`" -v --cov --cov-report=term` - doesn't work

`py.test test_things.py  --dist=load --tx="2*popen//python=`which python`" -v --cov` - works, and prints the terminal report.

So maybe it's an argument processing problem?

**Comment by ionelmc** [comment]
Yep, something really bad going on there. 

I'm working on a fix in #158.

**Comment by ionelmc** [comment]
Released `2.5.1`.

**Comment by tgs** [comment]
Thanks for the quick fix!
